detectors/resources/src/test/java/com/google/cloud/opentelemetry/detectors/GCEResourceTest.java

```diff
@@ -0,0 +1,68 @@
+package com.google.cloud.opentelemetry.detectors;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.junit.Assert.*;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.Map;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class GCEResourceTest {
+  @Rule public final WireMockRule wireMockRule = new WireMockRule(8089);
+  private final GCPMetadataConfig metadataConfig = new GCPMetadataConfig("http://localhost:8089/");
+  private final GCEResource testResource = new GCEResource(metadataConfig);
+
+  // Helper method to help stub endpoints
+  private void stubEndpoint(String endpointPath, String responseBody) {
+    stubFor(
+        get(urlEqualTo(endpointPath))
+            .willReturn(
+                aResponse()
+                    .withStatus(200)
+                    .withHeader("Content-Type", "text/plain")
+                    .withBody(responseBody)));
+  }
+
+  @Test
+  public void testGCEResourceWithEmptyAttributesReturnsEmpty() {
```

Comment:
As far as I understand, this tests validates behaviour when application is not running on GCP, so the test name should reflect that, e.g. `testGCEResourceNotGCP`.